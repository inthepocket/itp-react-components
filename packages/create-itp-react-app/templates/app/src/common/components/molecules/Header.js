import React from 'react';
import { string } from 'prop-types';
import Title from '../atoms/Title';
import styles from './Header.module.css';

const Header = ({ title }) => (
  <div className={styles.header}>
    <Title text={title} />
  </div>
);

Header.propTypes = {
  title: string,
};

Header.defaultProps = {
  title: 'Default Header',
};

export default Header;
