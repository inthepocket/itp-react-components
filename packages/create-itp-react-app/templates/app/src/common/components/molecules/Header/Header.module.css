@import '@inthepocket/itp-css/properties/breakpoints.css';

.header {
  position: relative;
  z-index: 1;
  padding: 10% 0 0 0;
  color: var(--colorSecondary900);
}

@media (--bp3) {
  .header {
    padding: 10% 0 0 7.5rem;
    width: 55vw;
  }
}
