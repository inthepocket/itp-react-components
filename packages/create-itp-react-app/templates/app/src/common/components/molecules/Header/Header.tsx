import * as React from 'react';
import Title from '<PROJECT-NAME>-common/components/atoms/Title/Title';
import { default as styles } from './Header.module.css';

const Header = ({
  title,
}: HeaderProps) => (
  <div className={styles.header}>
    <Title text={title} />
  </div>
);

interface HeaderProps {
  title: string;
}

export default Header;
