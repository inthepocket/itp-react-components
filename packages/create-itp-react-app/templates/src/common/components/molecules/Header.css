.header {
  background-color: var(--tertiaryColor);
  border-bottom: 5px solid var(--primaryColor);
  color: var(--secondaryColor);
  padding: 50px;
  text-align: center;
}
