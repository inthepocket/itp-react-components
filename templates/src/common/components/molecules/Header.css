.header {
  background-color: #10069f;
  color: #fff;
  padding: 50px;
  text-align: center;
}
