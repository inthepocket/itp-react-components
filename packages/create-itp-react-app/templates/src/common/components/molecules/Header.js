import React from 'react'
import {string} from 'prop-types'
import Title from 'itp-react-demo-common/components/atoms/Title'
import './Header.css'

const Header = ({title}) => (
  <div className="header">
    <Title text={title} />
  </div>
)

Header.propTypes = {
  title: string,
}

Header.defaultProps = {
  title: 'Default Header',
}

export default Header
