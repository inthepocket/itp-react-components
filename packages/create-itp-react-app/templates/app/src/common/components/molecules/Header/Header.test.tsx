import * as React from 'react';
import reactTestRenderer from 'react-test-renderer';
import Header from './Header';

test('Header snapshot: default', () => {
  const tree = reactTestRenderer
    .create((
      <Header title="foo:bar" />
    ))
    .toJSON();

  expect(tree).toMatchSnapshot();
});
