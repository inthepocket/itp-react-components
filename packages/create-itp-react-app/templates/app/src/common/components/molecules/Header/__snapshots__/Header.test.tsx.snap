// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header snapshot: default 1`] = `
<div
  className="header"
>
  <h1
    className="Title"
  >
    foo:bar
  </h1>
</div>
`;
